### DISCUSSION TITLE
Using ConPTY on Windows

### DISCUSSION THREAD

**Discussion by dzhus** [magit/magit#4894] [Ideas]
Hi all,

Like many I've just ran into https://github.com/magit/magit/issues/255.
Having read through https://github.com/magit/magit/wiki/Pushing-with-Magit-from-Windows#introduction I was surprised to discover that the root issue is

> Windows does not have the concept of a [PTY](https://en.wikipedia.org/wiki/Pseudo_terminal), so there is no way for Emacs to intercept password prompts from Git. You will get an error message like

This got me thinking about what would an ideal solution be.

Does ConPTY https://devblogs.microsoft.com/commandline/windows-command-line-introducing-the-windows-pseudo-console-conpty/ look like Windows infrastructure that could be used to solve the problem? If Emacs had bindings for ConPTY, would we be able to use them to add Windows-specific support to magit for this feature? (presumably with some changes to https://github.com/magit/magit/blob/b926ab0e051e497e8ed43754cfd2012d23dbfc4c/lisp/magit-process.el#L548, or even to the underlying process helpers in Emacs core).

**Comment by tarsius** [comment]
Possibly, but I don't use Windows, so someone else would have to figure that out and implement it.
